channeldb/invoices.go

```diff
@@ -494,19 +512,47 @@ func (d *DB) QueryInvoices(q InvoiceQuery) (InvoiceSlice, error) {
 			// Skip any settled invoices if the caller is only
 			// interested in unsettled.
 			if q.PendingOnly && invoice.Terms.Settled {
-				continue
-			}
-			// If we're not yet past the user defined offset, then
-			// we'll continue to seek forward.
-			if invoicesToSkip > 0 {
-				invoicesToSkip--
-				continue
+				return nil
 			}
 
 			// At this point, we've exhausted the offset, so we'll
 			// begin collecting invoices found within the range.
-			resp.Invoices = append(resp.Invoices, &invoice)
-			invoiceOffset++
+			resp.Invoices = append(resp.Invoices, invoice)
+			return nil
+		}
+
+		// If we know that a set of invoices exists, then we'll begin
+		// our seek through the bucket in order to satisfy the query.
+		// We'll continue until either we reach the end of the range, or
+		// reach our max number of invoices.
+	out:
+		for invoiceKey != nil {
+			err := forEachInvoice(invoiceKey)
+			switch {
+			case err == errDone:
+				break out
+			case err == nil:
+				if q.Reversed {
+					_, invoiceKey = cursor.Prev()
+				} else {
+					_, invoiceKey = cursor.Next()
+				}
+				continue
+			default:
+				return err
+			}
+		}
+
+		// If we iterated through the add index in reverse order, then
+		// we'll need to reverse the slice of invoices to return them in
+		// forward order.
+		if q.Reversed {
```

Comment:
I imagine when the `reversed` flag is provided, one assumes the results will be returned in reverse order also?

Reply:
I think it still makes sense to provide them in order. Perhaps it should be renamed to `backwards`?

Reply:
Callers can easily reverse themselves, so guess it's just a subjective preference.